src/lib.rs

```diff
@@ -12,19 +12,14 @@ pub struct LinkDrop {
     pub accounts: Map<PublicKey, Balance>,
 }
 
-const ACCESS_KEY_ALLOWANCE: u128 = 100_000_000_000_000_000_000;
-
-fn str_to_public_key(public_key: String) -> PublicKey {
-    let mut pk = vec![0];
-    pk.extend(bs58::decode(public_key).into_vec().unwrap());
-    pk
-}
+const ACCESS_KEY_ALLOWANCE: u128 = 1_000_000_000_000_000_000_000;
 
 #[near_bindgen]
 impl LinkDrop {
     /// Allows given public key to claim sent balance.
-    pub fn send(&mut self, public_key: String) -> Promise {
-        let pk = str_to_public_key(public_key);
+    #[payable]
+    pub fn send(&mut self, public_key: Base58PublicKey) -> Promise {
+        let pk = public_key.into();
         self.accounts.insert(&pk, &env::attached_deposit());
```

Comment:
What if there is already such public key? Another person can override this PK and remove the old balance. You need to increase the balance for this account instead.